BackEnd/src/main/java/com/unosquare/admin_core/back_end/controller/BaseController.java

```diff
@@ -51,12 +54,52 @@
                 EventDTO newEvent = modelMapper.map(date, EventDTO.class);
                 modelMapper.map(createEventViewModel, newEvent);
                 newEvent.setEventTypeId(eventType.getEventTypeId());
-                eventDTOS.add(newEvent);
+                if (eventType.getEventTypeId() == EventTypes.ANNUAL_LEAVE.getEventTypeId()) {
+                    UUID groupId = UUID.randomUUID();
+                    newEvent.setGroupId(groupId);
+                    splitEventIfFallsOnAWeekend(newEvent, newEvent.getEndDate(), eventDTOS);
+                } else {
+                    eventDTOS.add(newEvent);
+                }
             }
-
             eventService.saveEvents(eventDTOS.toArray(new EventDTO[0]));
         }
 
         return responses;
     }
+
+    private void splitEventIfFallsOnAWeekend(EventDTO event, LocalDate originalEndDate, List<EventDTO> eventDTOS) {
+        List<LocalDate> dateRange = getDatesRange(event.getStartDate(), event.getEndDate());
+        for (LocalDate eventDate : dateRange) {
+            if (eventDate.getDayOfWeek() == DayOfWeek.SATURDAY) {
```

Comment:
Should this if statement cover Sundays also?

Reply:
This logic does make sense, but it might need cleaned up a little. Let's grab some time on that.